src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -103,8 +112,48 @@ protected internal override PinMode GetPinMode(int pinNumber)
 
         protected internal override bool IsPinModeSupported(int pinNumber, PinMode mode)
         {
-            // Libgpiod Api do not support pull up or pull down resistors for now.
-            return mode != PinMode.InputPullDown && mode != PinMode.InputPullUp;
+            switch (mode)
+            {
+                case PinMode.Input:
+                    return true;
+                case PinMode.InputPullDown:
+                case PinMode.InputPullUp:
+                    // for use the bias flags we need libgpiod version 1.5 or later
+                    IntPtr libgpiodVersionPtr = Interop.libgpiod.gpiod_version_string();
+                    var libgpiodVersionMatch = Marshal.PtrToStringAnsi(libgpiodVersionPtr).Trim().Split('.');
+                    var libgpiodVersion = 0.0F;
+                    bool isLibgpiod1dot5 = false;
+
+                    if (libgpiodVersionMatch.Length > 1)
+                    {
+                        libgpiodVersion = float.Parse($"{libgpiodVersionMatch[0]}.{libgpiodVersionMatch[1]}");
+                        isLibgpiod1dot5 = (libgpiodVersion >= 1.5);
+                    }
+
+                    // for use the bias flags we need Kernel version 5.5 or later
+                    var linuxVersionMatch = RuntimeInformation.OSDescription.Replace("Linux ", "").Trim().Split('.');
```

Comment:
Can we re-use or copy following file to get the version instead? https://github.com/dotnet/runtime/blob/master/src/libraries/Common/tests/CoreFx.Private.TestUtilities/System/PlatformDetection.Unix.cs

Reply:
Hi @krwq, thanks for the review. I don't think so, this appears to check the version of the Distro and not the Kernel version. Here we have to check the Kernel version. If I'm wrong or if there is another way to get this kernel version more easily let me know.